src/System.CommandLine.Subsystems.Tests/VersionFunctionalTests.cs

```diff
@@ -0,0 +1,206 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Reflection;
+using FluentAssertions;
+using Xunit;
+using System.CommandLine.Parsing;
+
+namespace System.CommandLine.Subsystems.Tests
+{
+    public class VersionFunctionalTests
+    {
+        private static readonly string? version = (Assembly.GetEntryAssembly() ?? Assembly.GetExecutingAssembly())
+                                                 ?.GetCustomAttribute<AssemblyInformationalVersionAttribute>()
+                                                 ?.InformationalVersion;
+        private readonly string newLine = Environment.NewLine;
+
+        [Fact]
+        public void When_the_version_option_is_specified_then_the_version_is_written_to_standard_out()
+        {
+            var configuration = new CliConfiguration(new CliRootCommand());
+            var pipeline = new Pipeline();
+            var consoleHack = new ConsoleHack().RedirectToBuffer(true);
+            pipeline.Version = new VersionSubsystem();
+
+            var exit = pipeline.Execute(configuration, "-v", consoleHack);
+
+            exit.ExitCode.Should().Be(0);
+            exit.Handled.Should().BeTrue();
+            consoleHack.GetBuffer().Should().Be($"{version}{newLine}");
+        }
+
+        // TODO: invocation/output
+        /*
+                [Fact]
+                public async Task When_the_version_option_is_specified_then_invocation_is_short_circuited()
```

Comment:
I think this file was not recognized as a move because the commenting changed the indentation level